scripts/build/Jenkins/Jenkinsfile

```diff
@@ -384,6 +360,190 @@ def HandleDriveMount(String snapshot, String repositoryName, String projectName,
     }
 }
 
+//this should assume it will always be executed from the workspace root
+def ExecuteControlFile(controlFile, repoName) {
+    echo "${repoName}: ExecuteControlFile(${controlFile})"
+
+    PROCESSED_REPOS.add(repoName)
+
+    def ext = ''
+    if(env.IS_UNIX) {
+        ext = '.sh'
+    }
+
+    dir(ENGINE_REPOSITORY_NAME) {
+        //register all the objects specified in this controlFile
+        if(controlFile.REGISTER_ENGINES) {
+            echo 'REGISTER_ENGINES'
+            controlFile.REGISTER_ENGINES.each { engineName ->
+                PlatformSh("scripts/o3de${ext} register --engine-path ${workspace}/${repoName}/${engineName} || ver>nul", "Registering engine in ${workspace}/${repoName}/${engineName}")
+            }
+        }
+        if(controlFile.REGISTER_PROJECTS) {
+            echo 'REGISTER_PROJECTS'
+            controlFile.REGISTER_PROJECTS.each { projectName ->
+                PlatformSh("scripts/o3de${ext} register --project-path ${workspace}/${repoName}/${projectName} || ver>nul", "Registering project in ${workspace}/${repoName}/${projectName}")
+            }
+        }
+        if(controlFile.REGISTER_GEMS) {
+            echo 'REGISTER_GEMS'
+            controlFile.REGISTER_GEMS.each { gemName ->
+                PlatformSh("scripts/o3de${ext} register --gem-path ${workspace}/${repoName}/${gemName} || ver>nul", "Registering gem in ${workspace}/${repoName}/${gemName}")
+            }
+        }
+        if(controlFile.REGISTER_TEMPLATES) {
+            echo 'REGISTER_TEMPLATES'
+            controlFile.REGISTER_TEMPLATES.each { templateName ->
+                PlatformSh("scripts/o3de${ext} register --template-path ${workspace}/${repoName}/${templateName} || ver>nul", "Registering template in ${workspace}/${repoName}/${templateName}")
+            }
+        }
+        if(controlFile.REGISTER_RESTRICTED) {
+            echo 'REGISTER_RESTRICTED'
+            controlFile.REGISTER_RESTRICTED.each { restrictedName ->
+                PlatformSh("scripts/o3de${ext} register --restricted-path ${workspace}/${repoName}/${restrictedName} || ver>nul", "Registering restricted in ${workspace}/${repoName}/${restrictedName}")
+            }
+        }
+        //enable all gems specified in the control file
+        if(controlFile.ENABLE_GEMS) {
+            echo 'ENABLE_GEMS'
+            controlFile.ENABLE_GEMS.each { gemName ->
+                PlatformSh("scripts/o3de${ext} enable-gem --gem-path ${workspace}/${repoName}/${gemName} --project-name AutomatedTesting || ver>nul", "Enabling ${workspace}/${repoName}/${gemName} for AutomatedTesting")
+            }
+        }
+    }
+
+    //check out all repos
+    if(controlFile.REPOS) {
+        echo 'REPOS'
+        controlFile.REPOS.each { repo ->
+            //stop cyclical repos
+            if(PROCESSED_REPOS.contains(repo.NAME)) {
```

Comment:
As discussed offline: We may need to verify this test against workspaces that are restored from snapshot. The o3de and possibly the o3de-extras folder will already exist at development on first run

Reply:
we also need to test 2 more cases:
1. singular repo with no .automatedtesting.json
2. a compound repo with no .automatedtesting.json